scheduler/generic_sched.go

```diff
@@ -87,6 +87,8 @@ type GenericScheduler struct {
 	ctx        *EvalContext
 	stack      *GenericStack
 
+	// followUpEvals are evals with WaitUntil set, which are delayed until that time
+	// before being rescheduled
```

Comment:
`rescheduled` sounds unclear to me - I believe it means the scheduler re-processes in this context, not necessarily that these evals are for scheduled allocations due to client loss/drain/etc.

Reply:
They're submitted to the worker via RPC, which goes through the `eval_endpoint`, raft, the fsm, state_store, and then `evalbroker.processEnqueue`, where it gets `delayHeap.Push`ed. `evalbroker.runDelayedEvalsWatcher` checks the head of the delay heap, and waits until the first eval is due to add it to the regular eval queue. `worker.run` gets it from the channel and creates a new scheduler to process it then.

The `followupEvals` are only used for these delays, which hold up all of the reschedule processing. After they're due, they may become blocked or otherwise stopped if the job is changed.

Does that make sense? The context isn't saved, they go all the way around the eval system. On client loss or drain, the node drain eval creates the plan that changes all the affected allocs to `lost`. If the `reschedule` rules don't prevent it, replacement allocs will also be in that plan request. If `reschedule` or `stop_after_client_disconnect` prevent creating an immediate replacement alloc, it's only in that case you get a `followupEval`.